src/components/TodoFilter.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import classNames from 'classnames';
+import { FilterValue } from '../types/FiterValue';
+
+type Props = {
+  setFilterValue: (filterValue: FilterValue) => void;
+  filterValue: FilterValue;
+};
+
+export const TodoFilter = ({ setFilterValue, filterValue }: Props) => {
+  return (
+    // eslint-disable-next-line react/jsx-filename-extension
+    <nav className="filter">
+      <a
+        href="#/"
+        className={classNames('filter__link', {
+          selected: filterValue === FilterValue.All,
+        })}
+        onClick={() => setFilterValue(FilterValue.All)}
```

Comment:
make one function factory in component which will be accept filter value and return arrow function.
call it in each link with proper value.
```
const handleFilter = (filterValue: FilterValue) => () => serFilterValue(filterValue)
```
```suggestion
        onClick={handleFilter(FilterValue.All)}
```